cranelift-wasm/src/sections_translator.rs

```diff
@@ -357,7 +357,26 @@ pub fn parse_name_section<'data>(
                     environ.declare_func_name(FuncIndex::from_u32(index), name);
                 }
             }
-            _ => continue,
+            wasmparser::Name::Local(local_names) => {
+                let mut func_local_reader = local_names.get_function_local_reader()?;
+                for _ in 0..func_local_reader.get_count() {
+                    let func_local_names = func_local_reader.read()?;
+                    let func_index = func_local_names.func_index;
```

Comment:
The [wasm spec](https://webassembly.github.io/spec/core/appendix/custom.html#name-maps) says that the `func_index` here (and below `local_index`) should be uniquely defined, i.e. they can't be used twice. Is this done by wasm-parser, or do we need to add our own checks here?

Reply:
cc @yurydelendik who might have an answer here. I've looked very quickly at wasmparser.rs, and I don't think it does it. It could also be that the spec is outdated.

Reply:
The wasmparser does not do validation of name section, and in most of the cases "like any custom section they may be ignored". Expect the name section to be invalid at any point, e.g. old version of name section format or non-unique names. There is no defined behavior for invalid or incomplete data in the section (possible behaviors are: reject entire section, or recover and use valid portions).

FWIW I know that only the function names entries are used at this point (e.g. LLVM or binaryen)